www/js/services/commHelper.ts

```diff
@@ -229,3 +229,32 @@ export function putOne(key, data) {
     throw error;
   });
 }
+
+export function getModes() {
+  return new Promise((rs, rj) => {
+    window['cordova'].plugins.BEMServerComm.getUserPersonalData('/mode/get', rs, rj);
+  }).catch((error) => {
+    error = 'While getting modes, ' + error;
+    throw error;
+  });
+}
+
+export function updateMode(oldMode, newMode, isNewModeMustAdded) {
```

Comment:
I would make these function names more descriptive so it's clear these are for user-defined modes only.

Potentially `getCustomModes`+`updateCustomMode`, or even `getUserCustomModes`+`updateUserCustomMode`

Reply:
~~I also see that the words here are `get`/`update` and I wonder if `get`/`put` would be more consistent with the terminology on the server~~

Reply:
^ Nevermind, after looking I agree that `update` is better